src/openforms/authentication/contrib/org_oidc/oidc_plugins/plugins.py

```diff
@@ -23,9 +23,6 @@ def get_schema(self) -> JSONObject:
     def handle_callback(self, request: HttpRequest) -> HttpResponse:
         return callback_view(request)  # pyright: ignore[reportReturnType] # .as_view() returns HttpResponseBase
 
-    def _get_legacy_callback(self) -> str:
```

Comment:
Since this class implements this protocol: https://github.com/open-formulieren/open-forms/blob/8dc49136586929e646349b267f8125dea75d72d0/src/openforms/contrib/auth_oidc/plugin.py#L140
Shouldn't this method be implemented?

Reply:
IMO this whole protocol inheritance chain is messy. It's hard to follow what comes from where, it breaks type checking w/r to the `__init__` method, the order of base classes goes from generic -> specific instead of the other way around and I think it needs to be addressed in the library first.

I also have the impression that you think protocols act like abstract base classes, but that's not the case. They essentially only exist for static code analysis (from a practical point of view).

A child class will not complain about a method not being implemented/defined, because the protocol already defines it. A protocol basically says to the type checker "trust me, this thing exists". If it doesn't and you don't call it, nothing will fall over.

Reply:
I think that what i am confused about is how to make it clear that a class implements a particular protocol, so that then you know which methods you can expect in a class :thinking: So I went with the inheritance thing, but are there better ways to achieve this?

I know it won't complain if you don't call the method, but I thought that since it is "expected" because you are implementing a protocol, then if in the future things changes and it does get called then there is a problem. In this case it doesn't really matter since this method is deprecated, but more for my understanding :sweat_smile: